packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -412,6 +443,9 @@ abstract class ImageStreamCompleter with Diagnosticable {
         callback();
       }
       _onLastListenerRemovedCallbacks.clear();
+      if (!keepAlive && _currentImage?.autoDispose == true) {
+        _currentImage!.image.dispose();
```

Comment:
What if the last listener is removed and then another one is added?

Reply:
It will only work if the imageCache still has a reference to the image.

I could add an assert about that.

Reply:
Is that guaranteed at that in such a scenario the image would still be in the cache, though?

Reply:
No, someone could have evicted it or just disabled the cache